app/models/comment.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+class Comment < ActiveRecord::Base
+  validates :body, presence: { message: 'of comment can not be empty' }
+
+  belongs_to :commentable, polymorphic: true
+  belongs_to :author, foreign_key: :user_id, class_name: 'User'
+  has_one    :rating, as: :rateable, dependent: :destroy
+  has_many   :downvoters, through: :rating, source: :downvoters # users, who decrease comment rating
+  has_many   :upvoters, through: :rating, source: :upvoters # users, who increase comment rating
+  has_many   :comments, as: :commentable
+
+  after_create :create_rating # metod aded by ActiveRecord has_one
+
+  def accessible_for?(user)
+    user_id == user.id && !user.anonimous?
+  end
+
+  def destroy
+    if comments.empty?
+      super
+    else
+      errors.add(:base, 'Comment with replys can not be deleted')
```

Comment:
`replies`, not `replys`.

Also, isn't this the same as `has_many :comments, dependent: :restrict_with_error`?
If not, can you please make this a custom validation running only before destroy, not a method overload (which you should try to avoid at all costs in ActiveRecord, things can really go south...)